plugins/org.eclipse.sirius.diagram.sequence/src/org/eclipse/sirius/diagram/sequence/business/internal/layout/horizontal/SequenceHorizontalLayout.java

```diff
@@ -633,6 +636,51 @@ private boolean layoutReflexiveMessages(Map<? extends ISequenceElement, Rectangl
         return applied;
     }
 
+    private boolean layoutInteractionContainer(Map<? extends ISequenceElement, Rectangle> bounds, boolean pack) {
+        boolean applied = false;
+        Set<InteractionContainer> allInteractionContainers = this.sequenceDiagram.getAllInteractionContainers();
+        if (!allInteractionContainers.isEmpty() && getInteractionContainerBounds() != null) {
+            // Reset width of the interaction container
+            Bounds interactionContainerBounds = getInteractionContainerBounds();
+            interactionContainerBounds.setWidth(InteractionContainer.DEFAULT_WIDTH);
+            // Check that all instance role are within bounds of the interaction container, resize it otherwise
+            for (InstanceRole instanceRole : Iterables.filter(bounds.keySet(), InstanceRole.class)) {
+                final Node instanceRoleNode = instanceRole.getNotationNode();
+                LayoutConstraint layoutConstraint = instanceRoleNode.getLayoutConstraint();
+                if (layoutConstraint instanceof Bounds instanceRoleBounds
+                        && instanceRoleBounds.getX() + instanceRoleBounds.getWidth() + InteractionContainer.MARGIN > interactionContainerBounds.getWidth()) {
+                    // The instance role east bound + margin is further away than the interaction container east bounds
+                    interactionContainerBounds.setWidth(instanceRoleBounds.getX() + instanceRoleBounds.getWidth() + InteractionContainer.MARGIN);
+                }
+                applied = true;
+            }
+
+            // Check that all lost message are within bounds of the interaction container, resize it otherwise
+            for (LostMessageEnd lostMessageEnd : Iterables.filter(bounds.keySet(), LostMessageEnd.class)) {
```

Comment:
Do not forget all cases : messageToSelf width, hierarchy of combined fragments.

I assume that you might simply iterate over all passed bounds map entries and perform a "max" operation to get the aggregated max right et max range